autogen/main/cluster.tf.tmpl

```diff
@@ -119,6 +121,25 @@ resource "google_container_cluster" "primary" {
   {% endif %}
 {% endif %}
   {% if autopilot_cluster != true %}
+  {% if beta_cluster != true %}
+  # only one of logging/monitoring_service or logging/monitoring_config can be specified
+  logging_service    = local.logmon_config_is_set ? null : var.logging_service
```

Comment:
We already have this in 
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/4bf0011202667b6efe8a0b5aea2910f1c3250968/autogen/main/cluster.tf.tmpl#L81-L105
We should just change the conditional.

Reply:
I wasn't sure about "cluster_telemetry_type_is_set", "managed_prometheus", and "monitoring_enable_managed_prometheus" as I was primarily looking for "logging_config", but I can revise the PR to include those as well if you think they are ready to promote out of beta @bharathkkb

Reply:
@bharathkkb, `cluster_telemetry` and `managed_prometheus` are still in beta as per below, so it does not look possible to simply modify the conditional. Please advise.
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster

Reply:
@rwong2888 Can we try doing it inline and guard the dynamic block for managed_prometheus with a conditional. Similarly for the `local.cluster_telemetry_type_is_set || local.logmon_config_is_set` conditional.